packages/control-group/src/styles.tsx

```diff
@@ -0,0 +1,271 @@
+    case 'equal':
+    default:
+      return 'inline-grid'
+  }
+}
+
+const getState = (_: ButtonProps & ThemeProps, ommitActive?: boolean) => {
+  if (_.disabled) {
+    return 'disabled'
+  }
+  if (_.invalid) {
+    return 'invalid'
+  }
+  if ((_.activeGroup || _.checked) && !ommitActive) {
+    return 'active'
+  }
+
+  return 'default'
+}
+
+const getButtonGroupStyle = (_: WrapperProps & ThemeProps) => {
+  const openStyle = () => {
+    const elemLiStyle = {}
+    if (!_.isOpen && _.visibleItemsCount) {
+      elemLiStyle[
+        `> button:nth-child(n + ${_.visibleItemsCount}):not(:nth-last-child(1))`
+      ] = {
+        display: 'none'
+      }
+    }
+
+    return elemLiStyle
+  }
+
+  return {
+    position: 'relative',
+    zIndex: 1,
+    display: 'flex',
+    flexDirection: _.isOpen ? 'column' : 'row',
+    margin: 0,
+    padding: 0,
+    maxWidth: '100%',
+
+    ...openStyle(),
+    borderRadius: _.theme.radius
+  }
+}
+
+const getHorizontalGroupStyle = (_: WrapperProps & ThemeProps) => ({
+  display: getDisplay(_.layout || 'equal'),
+  gridTemplateColumns: '1fr '.repeat(_.childrenAmount || 0),
+  gridColumnGap: _.theme.spacing.xxs
+})
+
+const getVerticalGroupStyle = (_: WrapperProps & ThemeProps) => ({
+  display: getDisplay(_.layout || 'equal'),
+  gridTemplateColumns: '1fr ',
+  gridRowGap: _.theme.spacing.xxs,
+
+  '> *': {
+    marginBottom: _.theme.spacing.xxxs
+  }
+})
+
+const getControlGroupStyle = (_: WrapperProps & ThemeProps) => {
+  switch (_.groupAppearance) {
+    case 'button':
+      return getButtonGroupStyle(_)
+    case 'radio-horizontal':
+    case 'checkbox-horizontal':
+      return getHorizontalGroupStyle(_)
+    case 'radio-vertical':
+    case 'checkbox-vertical':
+      return getVerticalGroupStyle(_)
+    default:
+      return {}
+  }
+}
+
+export const ControlGroupWrapper = styled.div<WrapperProps>`
+  ${_ => ({
+    ...getControlGroupStyle(_)
+  })}
+`
+
+const getBorder = (_: ThemeProps & ButtonProps) => {
+  const {outline} = _.theme.colors.button
+  const border = (state: keyof typeof outline.borderColor) =>
+    `${outline.borderWidth}px solid ${outline.borderColor[state]}`
+
+  const state = getState(_) as keyof typeof outline.borderColor
+  const borderStyle = border(state)
+
+  return {
+    borderTop: _.isOpen ? border(getState(_, true)) : borderStyle,
+    borderBottom: _.isOpen ? 'none' : borderStyle,
+    borderRight: _.isOpen ? borderStyle : border(getState(_, true)),
+    borderLeft: _.isOpen ? borderStyle : 'none',
+
+    '&:first-of-type': {
+      borderTop: borderStyle,
+      borderLeft: borderStyle
+    },
+    '&:nth-last-of-type(2)': {
+      borderBottom: borderStyle
+    },
+    '&:last-of-type': {
+      borderRight: _.isOpen ? 'none' : borderStyle
+    }
+  }
+}
+
+const getBorderRadius = (_: ThemeProps & {isOpen?: boolean}) => ({
+  borderRadius: 0,
+
+  '&:first-of-type': {
+    borderRadius: _.isOpen
+      ? `${_.theme.radius} ${_.theme.radius} 0 0`
```

Comment:
Minor bug occurs: when there are only 2 buttons in a group, first one have no border radius:
![Zrzut ekranu 2019-12-5 o 08 41 49](https://user-images.githubusercontent.com/8265332/70214133-26d19480-173b-11ea-83a8-106d50cd02a5.png)

Reply:
fixed